### ISSUE TITLE
[React 19]: `eslint-plugin-react-compiler` gives false positive if hook takes generics

### ISSUE THREAD

**Issue by SimenB** [facebook/react#29062]
## Summary

(Not sure if this should be filed under React 19?)

The following code triggers `Hooks may not be referenced as normal values, they must be called. See https://react.dev/reference/rules/react-calls-components-and-hooks#never-pass-around-hooks-as-regular-values`

```ts
interface Data {}

function useHello() {
  const queryClient = useQueryClient();

  return useMutation<Data>({});
}
```

(`useQueryClient` and `useMutation` coming from `@tanstack/react-query`, fwiw)

Either removing the `<Data>` generic, _or_ the preceding `useQueryClient` resolves the error.

This is using `eslint-plugin-react-compiler@0.0.0-experimental-e04a001-20240515`

---

Anyways, I'm _super_ excited that you've finally open sourced this, and look forward to using it!

(I do wonder why `pretty-format@24` is used in the babel plugin tho 👀 very old!)

**Comment by zolotoygagash** [comment]
ср, 15 мая 2024 г., 21:21 Simen Bekkhus ***@***.***>:

**Comment by josephsavona** [comment]
Thanks for posting this, we're looking into it.

**Comment by NMinhNguyen** [comment]
I can confirm that https://github.com/facebook/react/pull/29081 appears to have fixed this issue :)

**Comment by poteto** [comment]
yep this was just fixed in #29081, thanks for reporting!
